packages/compilers/src/lib/common.ts

```diff
@@ -74,9 +74,15 @@ export function asyncExec(
         if (error) {
           reject(error);
         } else if (stderr) {
-          reject(
-            new Error(`Compiler process returned with errors:\n ${stderr}`),
-          );
+          // Vyper compilers <0.4.0 outputs warnings to stderr
+          // we handle this by checking if the stderr starts with "Warning:"
+          if (stderr.startsWith('Warning:')) {
```

Comment:
Isn't stderr the complete output over multiple lines? This just checks the first line, but is there a possibility that the first line is a warning and the second an error?

Reply:
Good idea, let me check

Reply:
ah okay that makes sense :+1: